### PR TITLE
Fix `varint::encode` allocating for small inputs 

### PR DESCRIPTION
**Pull request by Vurich**
Fixes a `TODO` in `varint`

### CODE REVIEW

File: multistream-select/src/protocol/listener.rs [hunk 1]
```diff
@@ -113,11 +113,13 @@ impl<R> Sink for Listener<R>
 			}
 
 			ListenerToDialerMessage::ProtocolsListResponse { list } => {
+				use std::iter;
```
Comment (tomaka) [inline multistream-select/src/protocol/listener.rs]: I'm not sure what the best practice is, but I usually prefer to put everything at the top of the file.

File: varint-rs/src/lib.rs [hunk 2]
```diff
@@ -491,12 +489,14 @@ pub fn decode<R: Read, T: Default + DecoderHelper>(mut input: R) -> errors::Resu
 }
 
 /// Syncronously decode a number from a `Read`
-pub fn encode<T: EncoderHelper + Bits>(input: T) -> Vec<u8> {
+pub fn encode<T: EncoderHelper + Bits>(input: T) -> Bytes {
+    use tokio_io::io::AllowStdIo;
```
Comment (tomaka) [inline varint-rs/src/lib.rs]: Same here.

File: multistream-select/src/protocol/listener.rs (hunk 1 above)
Comment (Vurich) [reply multistream-select/src/protocol/listener.rs]: I try and make the scope the minimum possible without unnecessary duplication, but I can't find any consensus online on local imports. I don't think we `use std::iter` anywhere else in the file so I think it's fine here.

File: varint-rs/src/lib.rs (hunk 2 above)
Comment (Vurich) [reply varint-rs/src/lib.rs]: I would agree with changing the `std::iter` import but I think this is very much an implementation detail of this particular function, and the global imports shouldn't have to worry about it.

File: multistream-select/src/protocol/listener.rs (hunk 1 above)
Comment (tomaka) [reply multistream-select/src/protocol/listener.rs]: I agreed with this in the past, but the problem is that I noticed that over time you tend to put the same local import(s) at multiple local scopes simultaneously, and things turn into spaghetti.

Comment (Vurich) [reply multistream-select/src/protocol/listener.rs]: Fair enough, I'll change it

**Pull request merged**